### PR TITLE
Landing default 1076 with api integration for solutionlist

### PR DESCRIPTION
**Pull request by Arti-Khillare1**

### CODE REVIEW

File: projects/self-creation-portal/src/app/components/app-main-view/app-main-view.component.ts [hunk 1]
```diff
@@ -27,18 +28,18 @@ export class AppMainViewComponent {
...
   }
 
+  ngOnInit(){
+    this.getnavData()
+  }
   onButtonClick(buttonTitle: string) {
   }
+
+  async getnavData(){
+    this.sidenavData = await this.formService.getForm(SIDE_NAV_DATA);
```
Comment (ramkumar-ns-dev) [inline projects/self-creation-portal/src/app/components/app-main-view/app-main-view.component.ts]: why there is await here ?

File: projects/self-creation-portal/src/app/components/create-new/create-new.component.ts [hunk 2]
```diff
@@ -24,89 +28,41 @@ export class CreateNewComponent {
...
-      this.headerData =  this.resourceHeader;
-    }
-    else if (cardItem.title === 'OBSERVATION_WITH_RUBRIC') {
-      this.headerData = this.observationwithrubricsHeader;
-    }
-    else {
-        this.headerData = {};
-    }
+  async getsolutionList() {
+    this.resourceList = await this.formService.getForm(SOLUTION_LIST)
```
Comment (ramkumar-ns-dev) [inline projects/self-creation-portal/src/app/components/create-new/create-new.component.ts]: same here

File: projects/self-creation-portal/src/app/services/form/form.service.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+  constructor(private httpService:HttpProviderService) { }
+
+  // Getting form from api
+  async getForm(formBody: any): Promise<any> {
+    const config = {
+      url: "scp/v1/form/read",
+      payload: formBody,
+  };
+  let result = await lastValueFrom(
```
Comment (ramkumar-ns-dev) [inline projects/self-creation-portal/src/app/services/form/form.service.ts]: why async here

**Review by ramkumar-pacewisdom — CHANGES_REQUESTED** [review]

File: projects/self-creation-portal/src/app/services/form/form.service.ts [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  providedIn: 'root'
+})
+export class FormService {
+
+  constructor(private httpService:HttpProviderService) { }
+
+  // Getting form from api
+  getForm(formBody: any) {
+    const config = {
+      url: "scp/v1/form/read",
```
Comment (ramkumar-ns-dev) [inline projects/self-creation-portal/src/app/services/form/form.service.ts]: why the URL is here ?
Please add it in url.config.json and use it here

File: projects/self-creation-portal/src/app/services/form/form.service.ts [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+      payload: formBody,
+  };
+  return this.httpService.post(config.url, formBody).pipe(
+    map((result: any) => {
+      let formData = _.pick(result, ['meta.formsVersion', 'result.result']);
```
Comment (ramkumar-ns-dev) [inline projects/self-creation-portal/src/app/services/form/form.service.ts]: why are you using lodash here ? is lodash installed ?

**Review by ramkumar-pacewisdom — CHANGES_REQUESTED** [review]

**Review by ramkumar-pacewisdom — APPROVED** [review]

**Pull request merged**